compiler/luci-interpreter/src/kernels/Quantize.cpp

```diff
@@ -0,0 +1,230 @@
+  }
+
+  output()->resize(input()->shape());
+}
+
+void Quantize::execute() const
+{
+  switch (input()->element_type())
+  {
+    case loco::DataType::FLOAT32:
+    {
+      tflite::QuantizationParams op_params;
+      op_params.zero_point = output()->zero_point();
+      op_params.scale = output()->scale();
+      const auto input_data = getTensorData<float>(input());
+
+      switch (output()->element_type())
+      {
+        case loco::DataType::S8:
+        {
+          luci_interpreter_pal::Quantize(op_params, getTensorShape(input()), input_data,
+                                         getTensorShape(output()), getTensorData<int8_t>(output()));
+          break;
+        }
+        case loco::DataType::U8:
+        {
+          luci_interpreter_pal::Quantize(op_params, getTensorShape(input()), input_data,
+                                         getTensorShape(output()),
+                                         getTensorData<uint8_t>(output()));
+          break;
+        }
+        case loco::DataType::S16:
+        {
+          luci_interpreter_pal::Quantize(op_params, getTensorShape(input()), input_data,
+                                         getTensorShape(output()),
+                                         getTensorData<int16_t>(output()));
+          break;
+        }
+        default:
+          throw std::runtime_error("Unsupported type.");
+      }
+      break;
+    }
+    case loco::DataType::S16:
+    {
+      int32_t multiplier;
+      int32_t shift;
+
+      compute_shift_multiplier(static_cast<double>(input()->scale()),
+                               static_cast<double>(output()->scale()), &multiplier, &shift);
+
+      const auto input_shape = getTensorShape(input());
+      const auto output_shape = getTensorShape(output());
+
+      const auto input_data = getTensorData<int16_t>(input());
+
+      const auto size = tflite::MatchingFlatSize(input_shape, output_shape);
+
+      switch (output()->element_type())
+      {
+        case loco::DataType::S8:
+        {
+          luci_interpreter_pal::Requantize(input_data, size, multiplier, shift,
+                                           input()->zero_point(), output()->zero_point(),
+                                           getTensorData<int8_t>(output()));
+          break;
+        }
+        case loco::DataType::S16:
+        {
+          luci_interpreter_pal::Requantize(input_data, size, multiplier, shift,
+                                           input()->zero_point(), output()->zero_point(),
+                                           getTensorData<int16_t>(output()));
+          break;
+        }
+        case loco::DataType::S32:
+        {
+          luci_interpreter_pal::Requantize(input_data, size, multiplier, shift,
+                                           input()->zero_point(), output()->zero_point(),
+                                           getTensorData<int32_t>(output()));
+          break;
```

Comment:
We are support `S32` quantization? =0
I think better to remove this.

Reply:
Done